deepspeed/runtime/engine.py

```diff
@@ -1145,6 +1145,10 @@ def forward(self, *inputs, **kwargs):
         return loss
 
     def allreduce_gradients(self, bucket_size=MEMORY_OPT_ALLREDUCE_SIZE):
+        # Pass (PP) gas boundary flag to optimizer (required for zero)
+        self.optimizer.is_gradient_accumulation_boundary = self.is_gradient_accumulation_boundary(
```

Comment:
Could self.optimizer be an fp32 optimizer, our fp16 optimizer, or client optimizer? Is it safe to assume `self.optimizer.is_gradient_accumulation_boundary`?